### ISSUE TITLE
Spawn on a specific workspace

### ISSUE THREAD

**Issue by Myrl** [baskerville/bspwm#520]
I know this can be emulated using rules and quickly spawning then unbinding the rule, but is there another way to do this?

**Comment by neeasade** [comment]
> then unbinding the rule

why not just make the rule `--one-shot`?

**Comment by Myrl** [comment]
Oh cool. That's pretty close to what I need; a dedicated command would
probably be more robust though.

**Comment by neeasade** [comment]
> a dedicated command would probably be more robust though.

what would something like this look like? I'm not understanding how much more robust you can get ahead of time (if you need more currently you could look into external rules perhaps).

**Comment by Myrl** [comment]
bspc spawn command -d ^2 ?

On Jul 20, 2016 7:49 AM, "Nathan Isom" <notifications@github.com> wrote:
>>
>> a dedicated command would probably be more robust though.
>
> what would something like this look like? I'm not understanding how much
more robust you can get ahead of time (if you need more currently you could
look into external rules perhaps).

**Comment by neeasade** [comment]
so that would be the equivalent of `bspc rule -a \* -o desktop=^2`? are you looking for brevity?

**Comment by Myrl** [comment]
I believe that makes the next window that pops up move to desktop 2?

It'd be better if we can specify that "this process's window will go to desktop 2" since that'd be more robust in the case of 2 windows opening after one another.

**Comment by neeasade** [comment]
the `\*` in the above is window class name. 

eg `bspc rule -a Termite -o desktop=^2`, `bspc rule -a chromium -o desktop=^3`.

**Comment by Myrl** [comment]
I see, thanks. It'd still have been nice though if you could explicitly instruct bspwm to spawn a process and move all its windows to the certain desktop.

**Comment by neeasade** [comment]
I see. You could always make a wrapper around bspc and extend it to do the above.

**Comment by baskerville** [comment]
You might infer a causal relationship between a window and a process only if the *_NET_WM_PID* property is set on the window.

**Comment by MarcosCosmos** [comment]
I would also like something like this - I'd mainly use it for setting up a bunch of commonly used windows (mainly multiple firefox windows) automatically when I start BSPWM. - so if you have a workaround script of somekind I'd love to see it.

I've been having trouble getting this working (I saw mention of creating empty/placeholder windows elsewhere but I think the feature must have been removed? I can't seem to find a syntax for it that BSPC doesn't spit an error at)
